### PR TITLE
modulesync 4.0.0

### PR DESCRIPTION
**Pull request by bastelfreak**
modulesync 4.0.0

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
For some reason Actions didn't run.

**Review by ekohl — APPROVED** [review]
Also cleaned up `.sync.yml` now.

**Comment by bastelfreak** [comment]
unit tests didn't pass because of puppet-lint. all acceptance tests passed. fixed puppet-lint warnings.

**Comment by bastelfreak** [comment]
now includes https://github.com/voxpupuli/puppet-python/pull/585

for reasons I don't know, there was a typo in the datatype. but that's already in master

**Comment by bastelfreak** [comment]
related PR for the required unit test changes: https://github.com/voxpupuli/puppet-python/pull/580/files#diff-040b40dcb171acf3b7bf835832475bdb9acbcf4fdf53f15d6a8f4d9be7ec9269R38

**Comment by bastelfreak** [comment]
I propose: we merge this as is. afterwards we can make cwd optional again in another PR

File: spec/defines/requirements_spec.rb [hunk 1]
```diff
@@ -33,7 +35,8 @@
           let :params do
             {
               owner: 'bob',
-              group: 'bob'
+              group: 'bob',
+              'cwd': '/foo'
```
Comment (ekohl) [inline spec/defines/requirements_spec.rb:39]: Isn't this valid too?
```suggestion
              cwd: '/foo'
```

File: spec/defines/requirements_spec.rb [hunk 2]
```diff
@@ -42,8 +45,16 @@
...
 
       context 'default' do
+        let :params do
+          {
+            'cwd': '/foo'
+          }
+        end
         it { is_expected.to contain_file('/requirements.txt').with_owner('root').with_group('root') }
       end
+      describe 'without cwd' do
```
Comment (ekohl) [inline spec/defines/requirements_spec.rb:55]: I'd visually separate these
```suggestion

      describe 'without cwd' do
```

**Review by ekohl — COMMENTED** [review]
Small coding style nits, otherwise :+1:

**Pull request merged**